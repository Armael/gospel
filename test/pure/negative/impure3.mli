val f : int -> int
(*@ y = f x
    pure
    raises Not_found -> true *)
