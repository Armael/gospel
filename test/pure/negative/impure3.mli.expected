
*******************************
********** Parsed file ********
*******************************
val f : int -> int[@@gospel
                    {| y = f x
    pure
    raises Not_found -> true |}]

*******************************
****** GOSPEL translation *****
*******************************
(*@ open Gospelstdlib *)

val f : int -> int
(*@ y = f x
    with ...
     *)
File "impure3.mli", line 5, characters 37-38:
Error: Type checking error: a pure function cannot raise exceptions
